docs/advanced_reference/developer_tooling/dev_mode.md

```diff
@@ -1,64 +1,103 @@
 
-## Debugging in the cluster
+    ```bash
+    docker image ls | grep oracle_hardhat
+    ```
 
-When debugging deployments, it can be useful to have the option to spin up a hardhat node to enable debugging and testing of the issue within the cluster. First, fetch the service:
+3. **Prepare the keys file.** Within the service folder, prepare a JSON file `keys.json` containing the wallet address and the private key for each of the agents that make up the service.
 
-```bash
-autonomy fetch valory/oracle_hardhat --local --service
-cd oracle_hardhat
-```
+    ??? example "Example of a `keys.json` file"
 
-You now need to replace the override in the ```service.yaml``` file: change ```http://host.docker.internal:8545``` to ```http://hardhat:8545```.
+        Find below an example of the structure of a `keys.json` file.
 
-Then, build the image:
-```bash
-autonomy build-image
-```
+        <span style="color:red">**WARNING: Use this file for testing purposes only. Never use the keys or addresses provided in this example in a production environment or for personal use.**</span>
 
-Now, push the image  to make it accessible for the cluster to pull it. You can get the tag from the previous command:
-```bash
-docker image push <tag>
-```
+        ```json
+        [
+          {
+              "address": "0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65",
+              "private_key": "0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a"
+          },
+          {
+              "address": "0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc",
+              "private_key": "0x8b3a350cf5c34c9194ca85829a2df0ec3153be0318b5e2d3348e872092edffba"
+          },
+          {
+              "address": "0x976EA74026E726554dB657fA54763abd0C3a0aa9",
+              "private_key": "0x92db14e403b83dfe3df233f83dfa3a0d7096f21ca9b0d6d6b8d88b2b4ec1564e"
+          },
+          {
+              "address": "0x14dC79964da2C08b23698B3D3cc7Ca32193d9955",
+              "private_key": "0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356"
+          }
+        ]
+        ```
+
+4. **Build the deployment.** Within the service folder, execute the command below to build the service deployment in dev mode.
+
+    ```bash
+    autonomy deploy build keys.json --dev --packages-dir ~/git/open-autonomy/packages --open-autonomy-dir ~/git/open-aea/ --open-aea-dir ~/git/open-autonomy/
+    ```
+
+    You must modify the paths in the command above aproppriately, pointing to:
+
+    * the path to the local registry (/packages directory),
+    * the path to the local {{open_autonomy_repository}},
+    * the path to the local {{open_aea_repository}}.
+
+    This will create a deployment environment in dev mode within the `./abci_build` folder.
+
+5. **Run the service.** Navigate to the deployment environment folder (`./abci_build`) and run the deployment locally. You also need to create and set up the directories `persistent_data/logs` and `persistent_data/venvs`.
```

Comment:
```
You also need to create and set up the directories `persistent_data/logs` and `persistent_data/venvs`.
```

The user does not have to create these directories, they are created by the framework when building the dev mode setup

Reply:
fixed